packages/desktop-client/src/components/modals/EditRule.js

```diff
@@ -397,13 +396,18 @@ function newInput(item) {
 
 export function ConditionsList({
   conditions,
-  conditionFields,
   editorStyle,
   isSchedule,
   onChangeConditions,
 }) {
   function addCondition(index) {
-    let field = 'payee';
+    // (remove the inflow and outflow pseudo-fields since they’d be a pain to get right)
+    let fields = conditionFields.slice(0, -2).map(f => f[0]);
+    for (let cond of conditions) {
+      fields = fields.filter(f => f !== cond.field);
+    }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏absolute nitpick as the data size is very tiny, so feel free to ignore

Current time complexity is O(n^2). If you changed the lookup to a hashmap, you could get this down to O(n).

Reply:
I did it this way intentionally so that the fields are added in a defined priority order, which I don’t think you can do with a hash map?

Reply:
Something like this (haven't tested):

```
const hashmap = new Set(conditions.map(({ field }) => field));
const newFields = fields.filter(f => !hashmap.has(f));
```